### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by axfelix**

### CODE REVIEW

File: exercises/async-activity-completion/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Part A: Retrieving the Task Token
+
+1. Edit the `workflow.go` file. In the `Activity()` definition, add a call to `activity.GetInfo()` that returns an `activityInfo` object.
+2. From that object, extract `activityInfo.TaskToken`. To asynchronously complete an Activity, you need to store this token outside of this Workflow, so that you can call it from another Temporal Client. The most straightforward way to do this is to encode it to hexadecimal or base64 and log it to your terminal, so add a call like `logger.Info("Activity", "taskToken", hex.EncodeToString(taskToken))`.
+3. Save the file.
+
+## Part B: Set Your Activity to Return `ErrResultPending`
+
+1. Continue editing the same Activity definition in the `workflow.go` file. You need to add a `return` statement that returns a special kind of error, `activity.ErrResultPending`, that will Temporal that the Activity has not failed but will be completed asynchronously.
```
Comment (angelazhou32) [inline exercises/async-activity-completion/README.md]: "that will Temporal that the Activity has not failed...?"

File: exercises/async-activity-completion/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+2. Note that the Workflow's `StartToCloseTimeout` has been lenghtened to 300 seconds for this exercise. Activities can still time out if they are running in the background.
```
Comment (angelazhou32) [inline exercises/async-activity-completion/README.md]:
```suggestion
2. Note that the Workflow's `StartToCloseTimeout` has been lengthened to 300 seconds for this exercise. Activities can still time out if they are running in the background.
```

File: exercises/async-activity-completion/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Exercise 3: Asynchronous Activity Completion
+
+During this exercise, you will:
+
+- Retrieve a task token from your Activity execution
+- Throw an `ErrResultPending` to indicate that the Activity is waiting for an external completion.using
```
Comment (angelazhou32) [inline exercises/async-activity-completion/README.md]:
```suggestion
Throw an `ErrResultPending` to indicate that the Activity is waiting for an external completion.
```

File: exercises/querying-workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+   return "", err
+}
+```
+
+4. Finally, update `currentState` again after that block of code, to something like "waiting for signal". This is the state that the Workflow will be waiting at when we query it.
+5. Save the file.
+
+## Part B: Performing a Query from a Client
+
+In this part of the exercise, you will create another Temporal client that sends a Signal. `queryclient/main.go` currently contains a near-empty Temporal client -- it looks like a Starter or a Worker, but it does not register any Workflows or do anything. You will use this client to send a Query.
```
Comment (MasonEgger) [inline exercises/querying-workflows/README.md]:
```suggestion
In this part of the exercise, you will create another Temporal client that sends a Query. `queryclient/main.go` currently contains a near-empty Temporal client -- it looks like a Starter or a Worker, but it does not register any Workflows or do anything. You will use this client to send a Query.
```

**Review by MasonEgger — APPROVED** [review]
LGTM

File: exercises/async-activity-completion/README.md (hunk 1 above)
Comment (axfelix) [reply exercises/async-activity-completion/README.md]: Fixed to "that will tell Temporal," thanks!

**Pull request merged**